components/Favorites/ToggleFav.tsx

```diff
@@ -0,0 +1,94 @@
+import React, { useState } from 'react';
+
+const favLSKey = "favorites";
+
+/*
+export default function FavStatus( props ) {
+    
+    const [ isFav, toggleFav ] = useToggleMovieFav( props.id );
+
+}
+*/
+
+export default function useToggleMovieFav ( id: number ) {
+
+    const [ isFav, setIsFav ] = useState( () => {
+        // Return true if the id is found in localStorage favorites array
+        // Return false if localStorage favorites array doesn't exist, or if id is not found in it
+        return ( localStorage.getItem( favLSKey ) &&
+                JSON.parse( localStorage[ favLSKey ] ).indexOf( id ) != -1 );
+    } );
+
+    const toggleFav = () => {
+        setIsFav( !isFav );
+        
+        if ( localStorage.getItem( favLSKey ) ) {
```

Comment:
Why would you need to check the local storage on every toggle you have the state in isFav. all you need is to. update the local storage when toggleFav is called no?